packages/next/src/lib/typescript/writeAppTypeDeclarations.ts

```diff
@@ -55,6 +55,8 @@ export async function writeAppTypeDeclarations({
     )
   }
 
+  directives.push('/// <reference path="./.next/types/routes.d.ts" />')
```

Comment:
```suggestion
  directives.push('/// <reference path="${path.relative(absolutePathNextEnvDTS, absoluteDistDirTypesRoutesDTS}" />')
```

The dist dir needs to come from `next.config` as it can be customized